File: sort-mapper-example-java/src/main/java/kh/farrukh/sortmapperexamplejava/lesson/model/LessonDetailsResponseDTO.java
package kh.farrukh.sortmapperexamplejava.lesson.model;

import kh.farrukh.sortmapperexamplejava.course.model.CourseResponseDTO;
import kh.farrukh.sortmapperexamplejava.teacher.model.TeacherResponseDTO;

public record LessonDetailsResponseDTO(
    Long id,
    String title,
    String description,
    Long courseId,
    CourseResponseDTO course,
    Long helperTeacherId,
    TeacherResponseDTO helperTeacher
) {

}

File: sort-mapper-example/src/main/kotlin/kh/farrukh/sortmapperexample/lesson/model/LessonDetailsResponseDTO.kt
package kh.farrukh.sortmapperexample.lesson.model

import kh.farrukh.sortmapperexample.course.model.CourseResponseDTO
import kh.farrukh.sortmapperexample.teacher.model.TeacherResponseDTO

data class LessonDetailsResponseDTO(
    val id: Long,
    val title: String,
    val description: String?,
    val courseId: Long,
    val course: CourseResponseDTO,
    val helperTeacherId: Long?,
    val helperTeacher: TeacherResponseDTO?
)
